### DISCUSSION TITLE
std::ostream Question

### DISCUSSION THREAD

**Discussion by starlight-traveler** [odygrd/quill#494] [General]
Does Quill support putting the ConsoleSink to a stream other than stdout? The ConsoleSink class doesn't seem to expose that in the User'sAPI and unsure of where to look for it.

Simply a pointer in the right direction would be appreciated.

**Comment by odygrd** [comment]
ConsoleSink can output to stdout or stderr. For the desired functionality i would recommend that you provide your own sink and do what you need inside write_log function. You can derive from existing sink or the base sink.

there is an example here

https://github.com/odygrd/quill/blob/master/examples/user_defined_sink.cpp
